perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ConsentsManagerBlImpl.java

```diff
@@ -346,4 +358,137 @@ private boolean hasValidConsent(PerunSession sess, Member member, ConsentHub con
+			.stream()
+			.collect(groupingBy(Consent::getUserId, toMap(Consent::getStatus, Function.identity())));
+
+		for (Facility facility : facilities) {
+			facilityAssignedServices = getPerunBl().getServicesManagerBl().getAssignedServices(sess, facility);
+
+			// get facility-assigned resources grouped by their Vo
+			Map<Integer, List<Resource>> assignedResources = getAssignedResources(sess, facility)
+				.stream()
+				.collect(groupingBy(Resource::getVoId, toList()));
+
+			// evaluate consents Vo by Vo, for all Vo resources
+			for (List<Resource> voResources : assignedResources.values()) {
+				evaluateConsents(sess, consentHub, voResources, facilityAssignedServices, userIdToConsents);
+			}
+		}
+		return consentHub;
+	}
+
+	/**
+	 * Evaluates consents for resources from one vo! All resources must come from same Vo.
+	 *
+	 * @param sess session
+	 * @param voResources resources from one Vo
+	 * @param facilityAssignedServices all services that are assigned to single facility
+	 * @param usersConsents users consents
+	 */
+	private void evaluateConsents(PerunSession sess, ConsentHub consentHub, List<Resource> voResources, List<Service> facilityAssignedServices,
+								  Map<Integer, Map<ConsentStatus, Consent>> usersConsents) {
+
+		if (voResources == null || voResources.size() == 0) {
+			return;
+		}
+
+		Vo vo = getPerunBl().getResourcesManagerBl().getVo(sess, voResources.get(0));
+		List<Member> members;
+		try {
+			members = getPerunBl().getMembersManager().getMembers(sess, vo);
+		} catch (PrivilegeException e) {
+			throw new InternalErrorException("evaluateConsents - not sufficient privileges");
+		} catch (VoNotExistsException e) {
+			throw new InternalErrorException(e);
+		}
+
+		for (Resource resource : voResources) {
+
+			Set<Integer> assignedUsers = getAssignedUsers(sess, resource)
+				.stream()
+				.map(PerunBean::getId)
+				.collect(Collectors.toSet());
+
+			// get only members on given resource
+			List<Member> membersOnResource = members.stream()
+				.filter((member -> assignedUsers.contains(member.getUserId())))
+				.collect(Collectors.toList());
+
+			// all services assigned to given resource
+			List<Service> servicesOnResource = getPerunBl().getResourcesManagerBl().getAssignedServices(sess, resource);
+
+			// get only services assigned to given resource as well as to facility
+			List<Service> filteredServicesOnResource = facilityAssignedServices.stream()
+				.filter((servicesOnResource::contains))
+				.collect(Collectors.toList());
+
+			// For each resource, check consents service by service
+			for (Service service : filteredServicesOnResource) {
+				List<Member> filteredMembersOnResource;
+
+				// depending on service, use active (possible expired too) members assigned to given resource
+				if (service.isUseExpiredMembers()) {
+					filteredMembersOnResource = membersOnResource.stream()
```

Comment:
Okay, I expected it to work the other way, but this makes sense as well. I fixed it - now group status is used (in the first case, status is compared against both VALID and EXPIRED value despite the fact, that only these two values are possible for group status, I wanted to make sure that in case some new status will be added in the future, it will not affect this)